sources/packages/api/src/route-controllers/application/application.student.controller.ts

```diff
@@ -112,12 +122,54 @@ export class ApplicationController extends BaseController {
         `Application id ${applicationId} was not found.`,
       );
     }
+    // Check wether the selected location is designated or not.
+    // If selected location is not designated, then make the
+    // selectedLocation null
+    if (application.data?.selectedLocation) {
+      const designatedLocation =
+        await this.locationService.getDesignatedLocationById(
+          application.data.selectedLocation,
+        );
+      const selectedLocation = await this.locationService.getLocationById(
```

Comment:
Do we need this second query when `getDesignatedLocationById` already provides the location name?

Reply:
`getDesignatedLocationById ` will only have designated location, 133 is for both designated and not designated locations

Reply:
I see, so, instead of returning the location if it is designated, we could return the location and one more property that would indicate if the location is designated or not.
This would move the EXISTS condition from WHERE to SELECT.
Just a suggestion 😉